### PR TITLE
Mappings editor/support unknown type cr changes

### PR DESCRIPTION
**Pull request by sebelga**
Here are the changes I made on my local branch for the review of https://github.com/elastic/kibana/pull/62149

### CODE REVIEW

File: x-pack/plugins/index_management/public/application/components/mappings_editor/components/document_fields/fields/fields_list_item.tsx [hunk 1]
```diff
@@ -265,7 +265,7 @@ function FieldListItemComponent(
                         dataType: TYPE_DEFINITION[source.type].label,
                       },
                     })
-                  : TYPE_DEFINITION[source.type]?.label ?? TYPE_DEFINITION.other.label}
+                  : (TYPE_DEFINITION[source.type] ?? TYPE_DEFINITION.other).label}
```
Comment (jloleysens) [inline x-pack/plugins/index_management/public/application/components/mappings_editor/components/document_fields/fields/fields_list_item.tsx:268]: Nice!

File: x-pack/plugins/index_management/public/application/components/mappings_editor/lib/serializers.ts [hunk 2]
```diff
@@ -45,16 +45,19 @@ const runParametersDeserializers = (field: Field): Field =>
...
 
 export const fieldSerializer: SerializerFunc<Field> = (field: Field) => {
+  const { otherTypeJson, ...rest } = field;
+  const updatedField: Field = Boolean(otherTypeJson) ? { ...otherTypeJson, ...rest } : { ...rest };
+
   // If a subType is present, use it as type for ES
-  if ({}.hasOwnProperty.call(field, 'subType')) {
-    field.type = field.subType as DataType;
-    delete field.subType;
+  if ({}.hasOwnProperty.call(updatedField, 'subType')) {
```
Comment (jloleysens) [inline x-pack/plugins/index_management/public/application/components/mappings_editor/lib/serializers.ts:52]: Just curious, what is the reason we use `{}.hasOwnProperty.call` pattern here? Do we use prototypical inheritance on the field objects somewhere?

File: x-pack/plugins/index_management/public/application/components/mappings_editor/lib/utils.ts [hunk 3]
```diff
@@ -123,7 +123,7 @@ const replaceAliasPathByAliasId = (
 };
 
 export const getMainTypeFromSubType = (subType: SubType): MainType =>
-  SUB_TYPE_MAP_TO_MAIN[subType] as MainType;
+  (SUB_TYPE_MAP_TO_MAIN[subType] ?? 'other') as MainType;
```
Comment (jloleysens) [inline x-pack/plugins/index_management/public/application/components/mappings_editor/lib/utils.ts:126]: 👍

File: x-pack/plugins/index_management/public/application/components/mappings_editor/lib/serializers.ts (hunk 2 above)
Comment (loixlab) [reply x-pack/plugins/index_management/public/application/components/mappings_editor/lib/serializers.ts:52]: I just use this pattern everywhere 😄 It became a habit.

**Pull request merged**